docs/src/explanations/Connection-operators-.md

```diff
@@ -0,0 +1,364 @@
+ChiselStage.emitVerilog(new Wrapper)
+```
+### Conclusion:
+The connection above went smoothly with no errors, this goes to show `<>` will work as long as there is at least one directioned thing (IO or submodule's IO) to "fix" the direction.
+
+
+## Concept 6: `<>` and `:=` connect signals by field name.
+This experiment creates a MockDecoupledIO which has the same fields by name as a DecoupledIO. Chisel lets us connect it and produces the same verilog, even though MockDecoupledIO and DecoupledIO are different types.
+( Scastie link for the experiment:https://scastie.scala-lang.org/Uf4tQquvQYigZAW705NFIQ)
+
+```scala mdoc:silent:reset
+import chisel3._
+import chisel3.stage.ChiselStage
+import chisel3.util.DecoupledIO
+
+class MockDecoupledIO extends Bundle {
+  val valid = Output(Bool())
+  val ready = Input(Bool())
+  val bits = Output(UInt(8.W))
+}
+class Wrapper extends Module{
+  val io = IO(new Bundle {
+  val in = Flipped(new MockDecoupledIO())
+  val out = new MockDecoupledIO()
+  })
+  val p = Module(new PipelineStage)
+  val c = Module(new PipelineStage) 
+  // connect producer to I/O
+  p.io.a <> io.in
+  // connect producer  to consumer
+  c.io.a <> p.io.b
+  // connect consumer to I/O
+  io.out <> c.io.b
+}
+class PipelineStage extends Module{
+  val io = IO(new Bundle{
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+    val b = DecoupledIO(UInt(8.W))
+  })
+  io.a <> io.b
+}
+```
+Below we can see the resulting Verilog for this example:
+```scala mdoc
+ChiselStage.emitVerilog(new Wrapper)
+```
+And here is another experiment, where we remove one of the fields of MockDecoupledIO:
+( Scastie link for the experiment:https://scastie.scala-lang.org/ChtkhKCpS9CvJkjjqpdeIA)
+
+```scala mdoc:silent:reset
+import chisel3._
+import chisel3.stage.ChiselStage
+import chisel3.util.DecoupledIO
+
+class MockDecoupledIO extends Bundle {
+  val valid = Output(Bool())
+  val ready = Input(Bool())
+  //val bits = Output(UInt(8.W))
+}
+class Wrapper extends Module{
+  val io = IO(new Bundle {
+  val in = Flipped(new MockDecoupledIO())
+  val out = new MockDecoupledIO()
+  })
+  val p = Module(new PipelineStage)
+  val c = Module(new PipelineStage) 
+  // connect producer to I/O
+  p.io.a <> io.in
+  // connect producer  to consumer
+  c.io.a <> p.io.b
+  // connect consumer to I/O
+  io.out <> c.io.b
+}
+class PipelineStage extends Module{
+  val io = IO(new Bundle{
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+    val b = DecoupledIO(UInt(8.W))
+  })
+  io.a <> io.b
+}
+```
+Below we can see the resulting Verilog for this example:
+```scala mdoc:crash
+ChiselStage.emitVerilog(new Wrapper)
+```
+This one fails because there is a field `bits` missing.
+
+### Conclusion:
+It doesn't matter the order of the fields, they need to be the same.
+For `:=`, the Scala types do not need to match but all the signals on the LHS must be provided by the RHS or you will get a Chisel elaboration error. There may be additional signals on the RHS, these will be ignored.
```

Comment:
This conclusion is a bit confusing given that we have shown both `<>` and `:=` and that they are different.
```suggestion
For `:=`, the Scala types do not need to match but all the signals on the LHS must be provided by the RHS or you will get a Chisel elaboration error. There may be additional signals on the RHS, these will be ignored. For `<>`, the Scala types do not need to match, but all signals must match exactly between LHS and RHS. In both cases, the order of the fields does not matter.

```